File: src/main/java/me/nepnep/nepbot/message/command/commands/fun/Lmddgtfy.java
package me.nepnep.nepbot.message.command.commands.fun;

import me.nepnep.nepbot.message.command.ICommand;
import me.nepnep.nepbot.message.command.Category;
import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.Permission;
import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;

import java.util.List;

public class Lmddgtfy implements ICommand {
    @Override
    public void execute(List<String> args, GuildMessageReceivedEvent event) {
        StringBuilder query = new StringBuilder();
        EmbedBuilder builder = new EmbedBuilder();

        for (String arg : args) {
            query.append(" " + arg);
        }

        String queryFormatted = query.toString().replace(" ", "%20");

        builder.setDescription("[Result](https://lmddgtfy.net/?q=" + queryFormatted + ")");

        event.getChannel().sendMessage(builder.build()).queue();
    }
    @Override
    public String getInvoke() {
        return "lmddgtfy";
    }
    @Override
    public Permission requiredPermission() {
        return null;
    }
    @Override
    public Category getCategory() {
        return Category.FUN;
    }
    @Override
    public String getDescription() {
        return "Searches something for you on duck duck go: ;lmddgtfy <String query>";
    }
}

File: src/main/kotlin/me/nepnep/nepbot/message/command/commands/fun/Lmddgtfy.kt
package me.nepnep.nepbot.message.command.commands.`fun`

import dev.minn.jda.ktx.messages.Embed
import me.nepnep.nepbot.message.command.AbstractCommand
import me.nepnep.nepbot.message.command.Category
import net.dv8tion.jda.api.entities.channel.middleman.GuildMessageChannel
import net.dv8tion.jda.api.events.message.MessageReceivedEvent

class Lmddgtfy : AbstractCommand(
    "lmddgtfy",
    Category.FUN,
    "Searches something for you on duck duck go: ;lmddgtfy <String query>"
) {
    override suspend fun execute(args: List<String>, event: MessageReceivedEvent, channel: GuildMessageChannel) {
        val query = args.subList(0, args.size).joinToString(" ")
        val safe = query.replace(" ", "%20")

        channel.sendMessageEmbeds(Embed { description = "[Result](https://lmddgtfy.net/?q=$safe)" }).queue()
    }
}